src/fileprovider.cs

```diff
@@ -0,0 +1,301 @@
+﻿﻿//
+// FileProvider C# bindings
+//
+// Authors:
+//	Alex Soto  <alexsoto@microsoft.com>
+//
+// Copyright 2017 Xamarin Inc. All rights reserved.
+//
+
+#if XAMCORE_2_0
+
+using System;
+using XamCore.ObjCRuntime;
+using XamCore.CoreGraphics;
+using XamCore.Foundation;
+
+namespace XamCore.FileProvider {
+
+	[iOS (11,0)]
+	[ErrorDomain ("NSFileProviderErrorDomain")]
+	[Native]
+	public enum NSFileProviderError : nint {
+		NotAuthenticated = -1000,
+		FilenameCollision = -1001,
+		SyncAnchorExpired = -1002,
+		InsufficientQuota = -1003,
+		ServerUnreachable = -1004,
+		NoSuchItem = -1005
+	}
+
+	[iOS (11,0)]
+	enum NSFileProviderItemIdentifier {
+		// Lets follow swift names
+		[Field ("NSFileProviderRootContainerItemIdentifier")]
+		RootContainer,
+
+		[Field ("NSFileProviderWorkingSetContainerItemIdentifier")]
+		WorkingSet,
```

Comment:
Container is missing? We do have RootContainer, other, remove "Container" from the previous value.